src/overlays/actors/ovl_En_Heishi/z_en_heishi.c

```diff
@@ -30,33 +41,151 @@ const ActorInit En_Heishi_InitVars = {
+    EnHeishi* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 25.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gSoldierSkeleton, &gSoldierWave, this->jointTable,
+                       this->morphTable, HEISHI_LIMB_MAX);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->paramsCopy = this->actor.params;
+    this->yawTowardsPlayer = this->actor.world.rot.y;
+
+    if (this->paramsCopy == 0) {
+        this->shouldSetHeadRotation = 1;
+        if (!(gSaveContext.weekEventReg[63] & 0x80) && !((gSaveContext.day == 3) && gSaveContext.isNight)) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    } else {
+        this->colliderCylinder.dim.radius = 30;
+        this->colliderCylinder.dim.height = 60;
+        this->colliderCylinder.dim.yShift = 0;
+        if ((gSaveContext.weekEventReg[63] & 0x80) || ((gSaveContext.day == 3) && gSaveContext.isNight)) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    }
+
+    this->actor.targetMode = 6;
+    this->actor.gravity = -3.0f;
+    Collider_InitAndSetCylinder(globalCtx, &this->colliderCylinder, &this->actor, &sCylinderInit);
+    this->actor.flags |= ACTOR_FLAG_8000000;
+    EnHeishi_SetupIdle(this);
+}
+
+void EnHeishi_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnHeishi* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+}
+
+void EnHeishi_ChangeAnimation(EnHeishi* this, s32 animIndex) {
+    static AnimationHeader* sAnimations[] = {
+        &gSoldierStandHandOnHip, &gSoldierCheerWithSpear, &gSoldierWave, &gSoldierSitAndReach, &gSoldierStandUp,
+    };
+    static u8 sAnimModes[] = {
+        ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP,
+        ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP,
+        ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_LOOP,
+    };
+
+    this->animIndex = animIndex;
+    this->frameCount = Animation_GetLastFrame(sAnimations[this->animIndex]);
+    Animation_Change(&this->skelAnime, sAnimations[this->animIndex], 1.0f, 0.0f, this->frameCount,
+                     sAnimModes[this->animIndex], -10.0f);
+}
+
+void EnHeishi_SetHeadRotation(EnHeishi* this) {
+    s16 yawDiff = this->yawTowardsPlayer - this->actor.world.rot.y;
+    s32 absYawDiff = ABS_ALT(yawDiff);
+
+    this->headRotXTarget = 0;
+    if ((this->actor.xzDistToPlayer < 200.0f) && (absYawDiff < 0x4E20)) {
+        this->headRotXTarget = this->yawTowardsPlayer - this->actor.world.rot.y;
+        if (this->headRotXTarget > 0x2710) {
+            this->headRotXTarget = 0x2710;
+        } else if (this->headRotXTarget < -0x2710) {
+            this->headRotXTarget = -0x2710;
+        }
+    }
+}
+
+void EnHeishi_SetupIdle(EnHeishi* this) {
+    s8 animIndex = HEISHI_ANIMATION_STAND_HAND_ON_HIP;
+
+    EnHeishi_ChangeAnimation(this, animIndex);
+    this->unk278 = animIndex;
```

Comment:
Shouldn't this be called animIndex?

Reply:
There's already an animIndex on the header and this variable isn't used anywhere where I can confirm it's intended functionality. The animIndex that already exists is used in EnHeishi_ChangeAnimation, so I'm much more confident that it's the "true" animIndex.